src/com/google/enterprise/adaptor/fs/FileDelegate.java

```diff
@@ -28,7 +28,7 @@
 import java.nio.file.attribute.FileTime;
 import java.util.Iterator;
 
-interface FileDelegate {
+interface FileDelegate extends DirectoryStreamFactory {
```

Comment:
I don't like this asymmetry of the delegate being an implicit factory for newDirectoryStream, and then being wrapped as one for newDfsLinkStream.